Source/dthread.cpp

```diff
@@ -155,10 +155,10 @@ void __cdecl dthread_cleanup()
 			error_buf = TraceLastError();
 			TermMsg("dthread3:\n(%s)", error_buf);
 		}
-		CloseHandle(sghThread);
+		DestroyEvent(sghThread);

```

Comment:
Avoid altering files in Source, they should stay as close to upstream as possible to avoid merge conflicts at this point. Instead try to implement things in miniwin:
https://github.com/diasurgical/devilutionX/blob/master/SourceX/miniwin_io.cpp#L132